lib/factory/data.factory.ts

```diff
@@ -17,24 +17,38 @@ export class DataFactory {
     );
 
     return {
-      generate: (count: number): Record<string, FactoryValue>[] => {
+      generate: <T = void>(count: number, context?: T): Record<string, FactoryValue>[] => {
         const ret = Array<Record<string, FactoryValue>>();
         for (let i = 0; i < count; i++) {
-          ret.push(this.generate(properties));
+          ret.push(this.generate<T>(properties, context));
         }
         return ret;
       },
     };
   }
 
-  private static generate(
-    properties: PropertyMetadata[],
+  private static generate<T = void>(
+    properties: PropertyMetadata<T>[],
+    context?: T
   ): Record<string, FactoryValue> {
     return properties.reduce(
-      (r, p) => ({
-        [p.propertyKey]: typeof p.arg === 'function' ? p.arg(faker) : p.arg,
-        ...r,
-      }),
+      (r, p) => {
+        // the values generated by the factory
+        const factoryIntermediate = typeof p.arg === 'function' ? p.arg(faker, context) : null
+        // the final value put in the object
+        const factoryValue = factoryIntermediate ?? p.arg;
+        // reassign the context with the new values
+        context = { 
+          ...context, 
+          ...(factoryIntermediate ?? {}),
```

Comment:
The value of factoryIntermediate will not always be an object, it can be a primitive value such as string.
If we perform a spread operator when the value is `male`, the result will be `{0: 'm', 1: 'a', 2: 'l', 3: 'e'}` which will be added to the context.